model/build-processor/src/main/java/org/keycloak/models/map/processor/GenerateEntityImplementationsProcessor.java

```diff
@@ -74,20 +74,24 @@ public class GenerateEntityImplementationsProcessor extends AbstractGenerateEnti
     protected void afterAnnotationProcessing() {
         if (! autogenerated.isEmpty()) {
             try {
-                JavaFileObject file = processingEnv.getFiler().createSourceFile("org.keycloak.models.map.common.AutogeneratedCloners");
+                JavaFileObject file = processingEnv.getFiler().createSourceFile("org.keycloak.models.map.common.AutogeneratedClasses");
```

Comment:
This class is getting messy with 200 lines of `put` calls. I was thinking about whether it would be nicer to use the approach we use in HotRod to have a descriptor for each Model instead of `HasMap`s - `class -> something` we could have an implementation of something like [this](https://github.com/keycloak/keycloak/blob/main/model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/common/HotRodEntityDescriptor.java) generated.

Maybe we could then also refactor `DeepCloner` to not have all the maps, but it needs a little more thinking as we need to be able to rewrite defaults using the builder.

This is not a strong requirement, just wanted to ask what you think about this.

Reply:
IMO we should keep it as is for the moment, but we can revisit. If needed I'd prefer to have it done in a separate task in the future.